pageserver/src/tenant/remote_timeline_client/index.rs

```diff
@@ -88,6 +89,12 @@ pub struct IndexPart {
 
     #[serde(default)]
     pub(crate) lineage: Lineage,
+
+    // Added in version 6. None == AuxFilePolicy::Unspecified. Use Option wrapper to keep forward compatibility.
+    // This flag is controlled by TenantConf::switch_aux_file_policy. When the first aux file gets written, the aux file
+    // mode will be persisted in `index_part.json`, and `switch_aux_file_policy` will be ignored.
```

Comment:
```suggestion
    /// Describes the kind of aux files stored in the timeline.
    /// 
    /// The value is modified during file ingestion when the latest wanted value communicated via tenant config is applied if it is acceptable.
    /// A V1 setting after V2 files have been committed is not accepted.
    /// 
    /// None == AuxFilePolicy::Unspecified.
```

This is my current understanding of the semantics.

I removed the "Option wrapper" because `serde(default)` matters for *backward* compatibility, it means `Default::default()` will be used if the field is not in the input.

This is why I don't understand why not just implement

```rust
impl Default for AuxFilePolicy { fn default() -> Self { AuxFilePolicy::Unspecified } }
``` 

together with 

```rust
impl AuxFilePolicy { fn is_default(&self) { self == &AuxFilePolicy::default() } }
```

for the `skip_serializing_if`.

We will have to be equally careful around changing the meaning of `None` as we need to change the meaning of `AuxFilePolicy::default`, but at least the code for this would be in a single place.

Could you explain the *forward* compatibility however, I don't see problems with that?

Reply:
Perhaps the disconnect is in that I don't see how `None` is different from `V1`, for me that just seems extra state to be carrying around.

Now that the AuxFilePolicy finally has type level comment, I think it shouldn't be repeated here.

Reply:
Note that now we don't have `AuxFilePolicy::Unspecified` any more. `AuxFilePolicy` is used for both tenant config and persistent flag. It's weird to allow admins to set aux file policy to unspecified...?

And regarding `None` and `V1`. For tenant config, there's no difference. If `switch_aux_file_policy` is not set and the user writes an aux file to the storage, it defaults to v1. But for the persistent flag in `index_part.json`, it matters. `None` means that nothing has been written to the storage before, and `v1` means that there is at least one write to the v1 storage but no write to the v2 storage.